pkg/pipelinerunmetrics/metrics.go

```diff
@@ -409,6 +455,13 @@ func (r *Recorder) RunningPipelineRuns(lister listers.PipelineRunLister) error {
 					prsWaitResolvingPipelineRef++
 				}
 			}
+		} else {
+			// If all PipelineRuns for a Pipeline are completed then record it with 0
+			// set the key to 0 so that we do not record further completed PipelineRuns  for same Pipeline
```

Comment:
I think the comment is a bit misleading here, it's not always "PipelineRuns for a Pipeline" as it depends on the config. 
In case there are no running PipelineRuns for the specific key, set the metric value to 0 to ensure the metric is set for the key.

Reply:
// In case there are no running PipelineRuns for the **pipelineRunKey**, set the metric value to 0 to ensure the metric is set for the key.